eng/common/scripts/Update-ChangeLog.ps1

```diff
@@ -0,0 +1,119 @@
+# Note: This script will add or replace version title in change log
+
+# Parameter description
+# Version : Version to add or replace in change log
+# Unreleased: Default is true. If it is set to false, then today's date will be set in verion title. If it is True then title will show "Unreleased"
+# ReplaceLatestEntry: Replaces the latest changelog entry, including its content.
+
+param (
+  [Parameter(Mandatory = $true)]
+  [String]$Version,
+  [Parameter(Mandatory = $true)]
+  [String]$ServiceDirectory,
+  [Parameter(Mandatory = $true)]
+  [String]$PackageName,
+  [boolean]$Unreleased=$True,
+  [boolean]$ReplaceLatestEntry = $False,
+  [String]$ReleaseDate
+)
+
+if ($ReleaseDate -and ($Unreleased -eq $True)) {
+    LogError "Do not pass 'ReleaseDate' arguement when 'Unreleased' is true"
+    exit 1
+}
+
+. "${PSScriptRoot}\common.ps1"
+
+if ($ReleaseDate)
+{
+    try {
+        $ReleaseStatus = ([DateTime]$ReleaseDate).ToString($CHANGELOG_DATE_FORMAT)
+        $ReleaseStatus = "($ReleaseStatus)"
+    }
+    catch {
+        LogError "Invalid 'ReleaseDate'. Please use a valid date in the format '$CHANGELOG_DATE_FORMAT'"
+        exit 1
+    }
+}
+elseif ($Unreleased) {
+    $ReleaseStatus = $CHANGELOG_UNRELEASED_STATUS
+}
+else {
+    $ReleaseStatus = "$(Get-Date -Format $CHANGELOG_DATE_FORMAT)"
+    $ReleaseStatus = "($ReleaseStatus)"
+}
+
+if ($null -eq [AzureEngSemanticVersion]::ParseVersionString($Version))
+{
+    LogError "Version [$Version] is invalid. Please use a valid SemVer"
+    exit(0)
+}
+
+$PkgProperties = Get-PkgProperties -PackageName $PackageName -ServiceDirectory $ServiceDirectory
+$ChangeLogEntries = Get-ChangeLogEntries -ChangeLogLocation $PkgProperties.ChangeLogPath
+
+
+if ($ChangeLogEntries.Contains($Version))
+{
+    if ($ChangeLogEntries[$Version].ReleaseStatus -eq $ReleaseStatus)
```

Comment:
At some point we should set the indention level to 2 spaces like the other ps files.